lesson17/main.js

```diff
@@ -1 +1,145 @@
+const slides = [];
+const url = "https://mocki.io/v1/010175de-0176-440a-9f90-d4a7ca8010cc";
 
+function renderStatus(response) {
+  const errorMessage = document.createElement("p");
+  errorMessage.textContent = `${response.status}:${response.statusText}`;
+  document.body.appendChild(errorMessage);
+}
+
+function displayInfo(error) {
+  const errorMessage = document.createElement("p");
+  errorMessage.textContent = error;
+  document.body.appendChild(errorMessage);
+}
+
+function getZIndex(slide) {
+  return parseInt(slide.style.zIndex, 10);
+}
+
+function getCurrentSlide(slides) {
+  return slides.find((slide) => getZIndex(slide) === slides.length - 1);
+}
+
+function updateButton(slides) {
+  const currentSlide = getCurrentSlide(slides);
+  currentSlide.parentNode
+    .querySelector(".js-current")
+    .classList.remove("js-current");
+  currentSlide.classList.add("js-current");
+  currentSlide.parentNode
+    .querySelector(".js-hidden")
+    ?.classList.remove("js-hidden");
+  if (currentSlide === slides[0]) {
+    document.getElementById("prev").classList.add("js-hidden");
+  }
+  if (currentSlide === slides[slides.length - 1]) {
+    document.getElementById("next").classList.add("js-hidden");
+  }
+}
+
+function updateSlidesNumber(slides) {
+  const currentSlide = getCurrentSlide(slides);
+  document.getElementById("slidesNumber").textContent = `${
+    slides.indexOf(currentSlide) + 1
+  }/${slides.length}`;
+}
+
+function slidesMovePrev() {
+  for (const slide of slides) {
+    const slideZIndex = getZIndex(slide);
+    slide.style.zIndex = (slideZIndex + slides.length - 1) % slides.length;
+  }
+  updateButton(slides);
+  updateSlidesNumber(slides);
+}
+
+function slidesMoveNext() {
+  for (const slide of slides) {
+    const slideZIndex = getZIndex(slide);
+    slide.style.zIndex = (slideZIndex + 1) % slides.length;
+  }
+  updateButton(slides);
+  updateSlidesNumber(slides);
+}
+
+function renderData(images) {
+  const fragment = document.createDocumentFragment();
+  const carousel = document.createElement("section");
+  carousel.id = "carousel";
+  const slidesContainer = document.createElement("div");
+  slidesContainer.id = "carouselContainer";
+  slidesContainer.className = "js-slides-container";
+  for (const [index, image] of images.entries()) {
```

Comment:
IMO
Since images is an array, I thought forEach would be sufficient.

```suggestion
  images.forEach((image, index) => {
```

Reply:
Thank you sae-san :)
It's simple.
98f3a5ac66be45b3376e44b815dcf4fb77591ba8